CA1802: Broken tests

#### Analyzer package
Microsoft.QualityGuidelines.Analyzers

#### Analyzer
CA1802: UseLiteralsWhereAppropriateAnalyzer

#### Repro steps
There are two unit tests ([here](https://github.com/dotnet/roslyn-analyzers/blob/master/src/Microsoft.QualityGuidelines.Analyzers/UnitTests/UseLiteralsWhereAppropriateTests.cs#L53) and [here](https://github.com/dotnet/roslyn-analyzers/blob/master/src/Microsoft.QualityGuidelines.Analyzers/UnitTests/UseLiteralsWhereAppropriateTests.cs#L92)) for this analyzer that produce a compiler error.

```
roslyn-analyzers\src\Microsoft.QualityGuidelines.Analyzers\UnitTests\UseLiteralsWhereAppropriateTests.cs(46,13): Test0.cs(8,34): error CS0236: A field initializer cannot reference the non-static field, method, or property 'Class1.f5'
roslyn-analyzers\src\Microsoft.QualityGuidelines.Analyzers\UnitTests\UseLiteralsWhereAppropriateTests.cs(82,13): Test0.vb(12,41): error BC30369: Cannot refer to an instance member of a class from within a shared method or shared member initializer without an explicit instance of the class.
```

I'm in the process of cleaning all tests from unintentional compile errors. It's unclear to me if these are intended.

If they are, I would like to add a comment and exception for them. If not, I'd like to know how I can correct these.

@basoundr Can you help out?

--- a/src/Microsoft.QualityGuidelines.Analyzers/UnitTests/UseLiteralsWhereAppropriateTests.cs
+++ b/src/Microsoft.QualityGuidelines.Analyzers/UnitTests/UseLiteralsWhereAppropriateTests.cs
@@ -50,8 +50,14 @@ public class Class1
     static string f3, f4 = ""Message is shown only for f4""; // Not readonly
     readonly int f5 = 3; // Not static
     const int f6 = 3; // Is already const
-    static readonly int f7 = 8 + f5; // f5 is not a const
+    static int f9 = getF9();
+    static readonly int f7 = 8 + f9; // f9 is not a const
     static readonly string f8 = null; // null value
+
+    private static int getF9()
+    {
+        throw new System.NotImplementedException();
+    }
 }");
         }
 
@@ -89,10 +95,16 @@ Public Class Class1
     ReadOnly f5 As Integer = 3
     ' Is already Const
     Const f6 As Integer = 3
-    ' f5 is not a Const
-    Shared ReadOnly f7 As Integer = 8 + f5
+    Shared f9 As Integer = getF9()
+    ' f9 is not a Const
+    Shared ReadOnly f7 As Integer = 8 + f9
     ' null value
     Shared ReadOnly f8 As String = Nothing
+
+    Private Shared Function getF9() As Integer
+        Throw New System.NotImplementedException()
+    End Function
+
 End Class");
         }
 

--- a/src/Microsoft.QualityGuidelines.Analyzers/UnitTests/UseLiteralsWhereAppropriateTests.cs
+++ b/src/Microsoft.QualityGuidelines.Analyzers/UnitTests/UseLiteralsWhereAppropriateTests.cs
@@ -104,7 +104,6 @@ Public Class Class1
     Private Shared Function getF9() As Integer
         Throw New System.NotImplementedException()
     End Function
-
 End Class");
         }
 

